Dockerfile

```diff
@@ -10,25 +9,40 @@ ADD demo/inventory /runner/inventory
 ADD https://releases.ansible.com/ansible-runner/ansible-runner.el8.repo /etc/yum.repos.d/ansible-runner.repo
 RUN dnf install -y epel-release && \
     dnf install -y ansible-runner python3-pip sudo rsync openssh-clients sshpass glibc-langpack-en git && \
+    alternatives --set python /usr/bin/python3 && \
     pip3 install ansible && \
-    chmod +x /bin/tini /bin/entrypoint && \
+    chmod +x /bin/tini && \
     rm -rf /var/cache/dnf
 
 RUN useradd runner && usermod -aG root runner
 
+ADD utils/entrypoint.sh /bin/entrypoint
+RUN chmod +x /bin/entrypoint
+
 # In OpenShift, container will run as a random uid number and gid 0. Make sure things
 # are writeable by the root group.
-RUN mkdir -p /runner/inventory /runner/project /runner/artifacts /runner/.ansible/tmp && \
-	chmod -R g+w /runner && chgrp -R root /runner && \
-	chmod g+w /etc/passwd
+RUN for dir in \
+      /home/runner \
+      /home/runner/.ansible \
+      /runner \
+      /home/runner \
+      /runner/env \
+      /runner/inventory \
+      /runner/project \
+      /runner/artifacts ; \
+    do mkdir -m 0775 -p $dir ; chmod g+rw $dir ; chown -R runner:runner $dir ; done && \
+    for file in \
+      /home/runner/.ansible/galaxy_token \
+      /etc/passwd ; \
+    do touch $file ; chmod g+rw $file ; chgrp runner $file ; done
 
 VOLUME /runner
 
 ENV LANG=en_US.UTF-8
 ENV LANGUAGE=en_US:en
 ENV LC_ALL=en_US.UTF-8
 ENV RUNNER_BASE_COMMAND=ansible-playbook
-ENV HOME=/runner
+ENV HOME=/home/runner
```

Comment:
@matburt This isn't changing where the host-dirs are mounted, only changing $HOME. What are the specific problems you think this will cause? There were some really gnarly things going on here. I will revert this part and post the error.